store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -169,7 +170,7 @@ public CarbonReaderBuilder withHadoopConf(String key, String value) {
           reader.initialize(split, attempt);
           readers.add(reader);
         } catch (Exception e) {
-          reader.close();
+          CarbonUtil.closeStreams(readers.toArray(new RecordReader[0]));
```

Comment:
Why not loop and close each one in the `readers`?

Reply:
`CarbonUtil.closeStreams` will loop and close the readers. Calling this will save the loc (line of code)